src/Telegram/Provider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace SocialiteProviders\Telegram;
+
+use Illuminate\Support\Facades\Validator;
+use InvalidArgumentException;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'TELEGRAM';
+
+    /**
+     * @return array
+     */
+    public static function additionalConfigKeys()
+    {
+        return [
+            'bot',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        return null;
+    }
+
+    public function getButton()
+    {
+        $botname = $this->config['bot'];
+        $callbackUrl = $this->config['redirect'];
```

Comment:
```suggestion
        $callbackUrl = $this->getConfig('redirect');
```

Reply:
Can you tell me why `getConfig()`, here, would be preferable? I chose `config[]` for consistency

Reply:
Because it's using the abstracted version, in case how we load/store the config changes.

Reply:
Resolved in 296cea3